app/src/main/res/layout/main_screen_fragment.xml

```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:layout_marginHorizontal="@dimen/default_side_margin">
+
+    <androidx.constraintlayout.widget.Guideline
+        android:id="@+id/guideline_horizontal"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        app:layout_constraintGuide_percent="0.5" />
+
+    <androidx.constraintlayout.widget.Guideline
+        android:id="@+id/guideline_vertical"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:orientation="vertical"
+        app:layout_constraintGuide_percent="0.5" />
+
+    <com.google.android.material.button.MaterialButton
+        android:id="@+id/btn_add_category"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/small_margin"
+        android:textAppearance="@style/TextAppearance.Material3.BodyLarge"
+        android:text="@string/add_category_btn_text"
+        app:layout_constraintBottom_toTopOf="@id/guideline_horizontal"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toEndOf="@id/guideline_vertical"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintWidth_max="@dimen/max_constraint_width_for_buttons" />
+
+    <com.google.android.material.button.MaterialButton
+        android:id="@+id/btn_add_flashcard"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="@dimen/small_margin"
+        android:textAppearance="@style/TextAppearance.Material3.BodyLarge"
+        android:text="@string/add_flashcards_btn_text"
+        app:layout_constraintBottom_toTopOf="@id/guideline_horizontal"
+        app:layout_constraintEnd_toStartOf="@id/guideline_vertical"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintWidth_max="@dimen/max_constraint_width_for_buttons" />
+
+    <com.google.android.material.textfield.TextInputLayout
+        android:id="@+id/til_spinner_categories"
+        style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox.ExposedDropdownMenu"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="?attr/colorOnSecondary"
```

Comment:
Tak tego nie robimy ;) Żeby zmieniać kolor jakiegokolwiek widgetu w taki sposób (a tak w ogóle to `app:boxBackgroundColor="?attr/colorOnSecondary"`) musi być bardzo dobre uzasadnienie a tu takiego nie widzę. Jeśli chcesz, żeby ExposedDropdownMenu miały jasne tło (a faktycznie lepiej wygląda z jasnym :) ) to trzeba to zrobić na poziomie stylu tak jak to robimy z kolorami 
`<style name="Theme.FlashcardsApp" parent="Theme.Material3.Light.NoActionBar">
        <item name="colorPrimary">@color/md_theme_light_primary</item>`
albo tworzymy nowy styl na bazie `@style/Widget.Material3.TextInputLayout.OutlinedBox.ExposedDropdownMenu"` i jego używamy w xml. 
Generalnie lepiej to nadpisać w stylu jeśli się da. Do poczytania: https://github.com/material-components/material-components-android/blob/master/docs/components/Menu.md#exposed-dropdown-menus

Reply:
Dzięki za wytłumaczenie i podrzucenie artykułu, będę pamiętał na przyszłość 😄 Poprawione ☺️